content/blog/2017/2017-01-18-refactoring-conditional-to-pipeline.adoc

```diff
@@ -0,0 +1,475 @@
+
+=== Conditions
+The previous example showed the "Strings match" condition and it's Pipeline equivalent.
+Let's look at couple more interesting conditions and their Jenkins Pipeline equivalents.
+
+==== Boolean condition
+
+You might think that a boolean condition would be the simplest condition, but it isn't.
+Since it works with string values from tokens, the Conditional BuildStep plugin offers
+a number of ways to indicate true or false.
+Truth is a case insensitive match of one of the following:
+1 (the number one), Y, YES, T, TRUE, ON or RUN.
+
+Pipeline can duplicate these, but depending on the scenario we might consider
+whether a simpler expression would suffice.
+
+.Pipeline
+[pipeline]
+----
+// Declarative //
+when {
+    // case insensitive regular expression for truthy values
+    expression { return token ==~ /(?i)(Y|YES|T|TRUE|ON|RUN)/ }
+}
+steps {
+    /* step */
+}
+
+// Script //
+// case insensitive regular expression for truthy values
+if (token ==~ /(?i)(Y|YES|T|TRUE|ON|RUN)/) {
+    /* step */
+}
+----
+
+==== Logical "OR" of conditions
+
+This condition wraps other conditions.
+It takes their results as inputs and performs a logical "or" of the results.
+The `AND` and `NOT` conditions do the same, performing their respective operations.
+
+.Pipeline
+[pipeline]
+----
+// Declarative //
+when {
+    // A or B
+    expression { return A || B }
+}
+steps {
+    /* step */
+}
+
+// Script //
+// A or B
+if (A || B) {
+    /* step */
+}
+----
+
+=== Tokens
+
+Tokens can be considerably more work than conditions.
+There are more of them and they cover a much broader range of behaviors.
+The previous example showed one of the simpler cases, accessing a build parameter,
+where the token has a direct equivalent in Pipeline.
+However, many tokens don't have direct equivalents,
+some take a parameters (adding to their complexity),
+and some provide information that is simply not exposed in pipeline yet.
+So, determining how to migrate tokens needs to be done on case-by-case basis.
+
+Let's look at a few examples.
+
+==== "FILE" token
+
+*Expands to the contents of a file. The file path is relative to the build workspace root.*
+
+_${FILE,path="PATH"}_
+
+This token maps directly to the `readFile` step.
+The only difference is the file path for `readFile` is relative to the
+current working directory on the agent, but that is the workspace root by default.
+No problem.
+
+.Pipeline
+[pipeline]
+----
+// Declarative //
+when {
+    expression { return readFile('pom.xml').contains('mycomponent') }
+}
+steps {
+    /* step */
+}
+
+// Script //
+if (readFile('pom.xml').contains('mycomponent')) {
+    /* step */
+}
+----
+
+
+==== GIT_BRANCH
+
+*Expands to the name of the branch that was built.*
+
+*Parameters* (descriptions omitted): _all_, _fullName_.
+
+This information may or may not be exposed in Pipeline.  If you're using
+multibranch pipeline plugin `env.BRANCH_NAME` give similar basic information, but doesn't
```

Comment:
If you're using multibranch pipeline plugin `env.BRANCH_NAME` give similar
-> 
If you're using the link:Multibranch Pipeline plugin, `env.BRANCH_NAME` will give